src/js/components/form-elements/v2/FileSelect.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback } from 'react';
+
+import PropTypes from 'prop-types';
+import { useDropzone } from 'react-dropzone';
+
+import Button from 'components/form-elements/Button';
+import useTranslate from 'hooks/useTranslate';
+
+import './style.scss';
+
+const FileSelect = ({
+  height,
+  width,
+  dropzoneText,
+  buttonLabel,
+  className,
+  multiple,
+  maxFiles,
+  allowedExtensions,
+  ...fieldProps
+}) => {
+  const onDrop = useCallback((acceptedFiles) => {
+    fieldProps.onChange?.(multiple ? acceptedFiles : acceptedFiles[0]);
+  }, []);
+
+  const translate = useTranslate();
+
+  const getFileExtension = (file) => file.path.split('.')?.[1];
+
+  const validateFileType = (file) => {
+    if (!allowedExtensions.length || allowedExtensions.includes(getFileExtension(file))) {
+      return null;
+    }
+
+    return {
+      code: 'invalid-extension',
+      message: translate(
+        'react.default.error.invalidFileExtension.label',
+        `File extension should be one of: ${allowedExtensions.join(', ')}`,
+        [allowedExtensions.join(', ')],
+      ),
+    };
+  };
+
+  const {
+    getRootProps, getInputProps, open, acceptedFiles, fileRejections,
+  } = useDropzone({
+    onDrop,
+    noClick: true,
+    noKeyboard: true,
+    validator: validateFileType,
+    multiple,
+    maxFiles,
+  });
+
+  const mapFiles = (files) => files.map((file) => {
+    const data = file?.path ? file : file.file;
+    return (
+      <li key={data.path}>
+        {data.path}
+        {' '}
+        (
+        {data.size}
+        {' '}
+        bytes)
+        {file?.errors?.length ? (
+          <ul>
+            {file.errors.map((e) => (
+              <li key={e.code}>{e.message}</li>
+            ))}
+          </ul>
+        ) : null}
```

Comment:
I would maybe do something like:
```suggestion
        <div>
          <span className="mx-1" >{data.path}<span>
          <span className="mx-1" >({data.size})<span>
        </div>
        {!!file?.errors?.length ? (
          <ul>
            {file.errors.map((e) => (
              <li key={e.code}>{e.message}</li>
            ))}
          </ul>
        ) }
```

But current solution is fine

Reply:
@drodzewicz I think you skipped the fallback (`:`), or you meant to do it via `&&`. Honestly I'd avoid double negation, since it is usually confusing. 
I would prefer Alan's version.

Reply:
Sure, Idc